### PR TITLE
Linkage issue in windows

### PR DESCRIPTION
**Pull request by ssrlive**
It always tell me that
```
c:\sql>cargo build
   Compiling libsqlite3-sys v0.25.2 (c:\sql\libsqlite3-sys)
warning: cl : Command line warning D9027 : source file 'C:\Program Files\OpenSSL-Win64\lib\libcrypto.lib' ignored
   Compiling rusqlite v0.28.0 (c:\sql)
    Finished dev [unoptimized + debuginfo] target(s) in 5.11s
```
so I change the script.

I think it can fix issue #966 also

### CODE REVIEW

File: libsqlite3-sys/build.rs [hunk 1]
```diff
@@ -185,8 +185,7 @@ mod build_bundled {
                 // Windows without `-vendored-openssl` takes this to link against a prebuilt
                 // OpenSSL lib
                 cfg.include(inc_dir.to_string_lossy().as_ref());
-                let lib = lib_dir.join("libcrypto.lib");
-                cfg.flag(lib.to_string_lossy().as_ref());
+                println!("cargo:rustc-link-lib=dylib=libcrypto");
```
Comment (gwenn) [inline libsqlite3-sys/build.rs:188]: So there is no difference anymore between `is_windows` (lines 184:188) and `use_openssl` (lines 190:183) (except `=libcrypto` vs `=crypto`) ?
Could you please try to fuse them ?

Comment (ssrlive) [reply libsqlite3-sys/build.rs]: OK. changed.

**Comment by HuakunShen** [comment]
Does this enable `bundled-cipher` feature on windows? 
I am still getting `  = note: LINK : fatal error LNK1181: cannot open input file 'libcrypto.lib'` on windows

**Comment by ssrlive** [comment]
https://github.com/rusqlite/rusqlite/blob/eebbbb008633b36f2580b29501800fb8bc37e36a/libsqlite3-sys/build.rs#L152-L153

**Comment by HuakunShen** [comment]
> https://github.com/rusqlite/rusqlite/blob/eebbbb008633b36f2580b29501800fb8bc37e36a/libsqlite3-sys/build.rs#L152-L153
> 
> https://slproweb.com/products/Win32OpenSSL.html

@ssrlive Thank you for your response, but can you elaborate a  little more on this? I didn't understand how to use the code you mentioned.
Do you mean I need to install openssl from https://slproweb.com/products/Win32OpenSSL.html? 
And how to use the two lines you quoted?

**Comment by ssrlive** [comment]
image

image

**Comment by HKMV** [comment]
@ssrlive  Thank you so much, I feel that the successful compilation of the solution provided by you is provided that OPENSSL_LIB_DIR be specified to the following specific directory such as: lib\VC\x64\MDd.
However, I don't know what MD, MDd, MT, and MTd mean.

**Comment by ssrlive** [comment]
```
~ pwd
/media/q/c/OpenSSL-Win64/lib
~ tree .
├── capi.def
├── capi.lib
├── dasync.def
├── dasync.lib
├── ec_internal_test.lib
├── legacy.def
├── legacy.lib
├── libapps.lib
├── libcommon.lib
├── libcrypto.def
├── libcrypto.lib
├── libcrypto_static.lib
├── libdefault.lib
├── liblegacy.lib
├── libssl.def
├── libssl.lib
├── libssl_static.lib
├── libtestutil.lib
├── loader_attic.def
├── loader_attic.lib
├── openssl.lib
├── ossltest.def
├── ossltest.lib
├── padlock.def
├── padlock.lib
├── p_test.def
├── p_test.lib
└── VC
    ├── libcrypto64MDd.lib
    ├── libcrypto64MD.lib
    ├── libcrypto64MTd.lib
    ├── libcrypto64MT.lib
    ├── libssl64MDd.lib
    ├── libssl64MD.lib
    ├── libssl64MTd.lib
    ├── libssl64MT.lib
    └── static
        ├── libcrypto64MDd.lib
        ├── libcrypto64MD.lib
        ├── libcrypto64MTd.lib
        ├── libcrypto64MT.lib
        ├── libssl64MDd.lib
        ├── libssl64MD.lib
        ├── libssl64MTd.lib
        └── libssl64MT.lib
```

**Comment by HKMV** [comment]
> ```
> ~ pwd
> /media/q/c/OpenSSL-Win64/lib
> ~ tree .
> ├── capi.def
> ├── capi.lib
> ├── dasync.def
> ├── dasync.lib
> ├── ec_internal_test.lib
> ├── legacy.def
> ├── legacy.lib
> ├── libapps.lib
> ├── libcommon.lib
> ├── libcrypto.def
> ├── libcrypto.lib
> ├── libcrypto_static.lib
> ├── libdefault.lib
> ├── liblegacy.lib
> ├── libssl.def
> ├── libssl.lib
> ├── libssl_static.lib
> ├── libtestutil.lib
> ├── loader_attic.def
> ├── loader_attic.lib
> ├── openssl.lib
> ├── ossltest.def
> ├── ossltest.lib
> ├── padlock.def
> ├── padlock.lib
> ├── p_test.def
> ├── p_test.lib
> └── VC
>     ├── libcrypto64MDd.lib
>     ├── libcrypto64MD.lib
>     ├── libcrypto64MTd.lib
>     ├── libcrypto64MT.lib
>     ├── libssl64MDd.lib
>     ├── libssl64MD.lib
>     ├── libssl64MTd.lib
>     ├── libssl64MT.lib
>     └── static
>         ├── libcrypto64MDd.lib
>         ├── libcrypto64MD.lib
>         ├── libcrypto64MTd.lib
>         ├── libcrypto64MT.lib
>         ├── libssl64MDd.lib
>         ├── libssl64MD.lib
>         ├── libssl64MTd.lib
>         └── libssl64MT.lib
> ```

The directory structure on my side is different from yours, here's my structure:
lib
 └── VC
     └── x64
         ├── MD
         │   ├── capi.def
         │   ├── capi.lib
         │   ├── dasync.def
         │   ├── dasync.lib
         │   ├── ec_internal_test.lib
         │   ├── evp_extra_test2.lib
         │   ├── legacy.def
         │   ├── legacy.lib
         │   ├── libapps.lib
         │   ├── libcommon.lib
         │   ├── libcrypto.def
         │   ├── libcrypto.lib
         │   ├── libcrypto_static.lib
         │   ├── libdefault.lib
         │   ├── liblegacy.lib
         │   ├── libssl.def
         │   ├── libssl.lib
         │   ├── libssl_static.lib
         │   ├── libtestutil.lib
         │   ├── loader_attic.def
         │   ├── loader_attic.lib
         │   ├── openssl.lib
         │   ├── ossltest.def
         │   ├── ossltest.lib
         │   ├── padlock.def
         │   ├── padlock.lib
         │   ├── p_minimal.def
         │   ├── p_minimal.lib
         │   ├── p_test.def
         │   └── p_test.lib
         ├── MDd
         │   ├── libcrypto.def
         │   ├── libcrypto.lib
         │   ├── libcrypto_static.lib
         │   ├── libssl.def
         │   ├── libssl.lib
         │   └── libssl_static.lib
         ├── MT
         │   ├── libcrypto.def
         │   ├── libcrypto.lib
         │   ├── libcrypto_static.lib
         │   ├── libssl.def
         │   ├── libssl.lib
         │   └── libssl_static.lib
         └── MTd
             ├── libcrypto.def
             ├── libcrypto.lib
             ├── libcrypto_static.lib
             ├── libssl.def
             ├── libssl.lib
             └── libssl_static.lib

**Comment by ssrlive** [comment]
Perhaps you must change your environment variable `OPENSSL_LIB_DIR` to `c:\OpenSSL-Win64\lib\vc\x64\md\'.

Have you never done windows development? 😡

**Comment by HKMV** [comment]
> Perhaps you must change your environment variable `OPENSSL_LIB_DIR` to `c:\OpenSSL-Win64\lib\vc\x64\md'.
> 
> Have you never done windows development? 😡

I've been doing Windows and Server related development.,I'm now configured as you said.,It's normal.,The above recovery is just to tell you about this situation.。

**Comment by ssrlive** [comment]
滾 gun！

**Comment by HKMV** [comment]
> 滾 gun！

怎么还骂人呢？大家都是中国人，还是文明点好。

**Comment by ssrlive** [comment]
遇到蠢人，着不起那個急。

**Comment by HKMV** [comment]
> 遇到蠢人，着不起那個急。

其实我上面第一次恢复就已经解决问题了，是来感谢你的，你这倒好，翻译过来翻译过去一会儿怼起来了

**Comment by ssrlive** [comment]
結束了。

**Pull request merged**